clang/lib/Basic/Targets/OSTargets.h

```diff
@@ -790,7 +790,9 @@ template <typename Target>
 class LLVM_LIBRARY_VISIBILITY UEFITargetInfo : public OSTargetInfo<Target> {
 protected:
   void getOSDefines(const LangOptions &Opts, const llvm::Triple &Triple,
-                    MacroBuilder &Builder) const override {}
+                    MacroBuilder &Builder) const override {
+    Builder.defineMacro("__UEFI__");
```

Comment:
I wonder if this should be consistent with the Unix flavors, I don't know what the UEFI target does or if GCC or another other compiler already has a macro for this.
```suggestion
    DefineStd(Builder, "uefi", Opts);
```

Reply:
Probably, although UEFI spec-wise is similar to Windows afaict. I'm not sure which style we'd want. GCC does not support UEFI targets and prefer to do a Windows style toolchain.

Reply:
Personally I prefer DefineStd for this area where possible or we know we're not adding conflicting macros. What do other toolchains produce here?

Reply:
I've switched over to `DefineStd`.

Reply:
`DefineStd` will define `uefi` (in the GNU mode), `__uefi` and `__uefi__`. I don't think we should be defining any new symbols in the user's namespace, even in the GNU mode. I also don't think we should have more then one define, we do it in other targets for backwards compatibility, but there is no backwards compatibility for UEFI. Regarding upper case vs lower case, I have mild preference for the upper case since it's more consistent with Windows and UEFI is essentially a flavor of Windows, not UNIX or GNU. TL;DR I'd prefer using `Builder.defineMacro("__UEFI__");` as you had originally.